src/Gating/GateDataSetLoader.cs

```diff
@@ -22,7 +22,7 @@ public class GateDataSetLoader : ConfigurationDataSetLoader<GateDataSet>
 		/// <param name="gates">The gates.</param>
 		/// <param name="testGroups">The test groups.</param>
 		public GateDataSetLoader(ICache cache, IResourceMonitor resourceMonitor, IResource gates, IResource testGroups)
-			: base(cache, resourceMonitor)
+			: base(cache, resourceMonitor, new GateDataSetFactory(gates.Name, testGroups.Name))
```

Comment:
suggest using gates?.Name and testGroups?.Name to allow the Code.Expects in GateDataSetFactory to log the missing values instead of throwing here.